### PR TITLE
Default to 1 OpenMP thread for our unit tests

### PR DESCRIPTION
**Pull request by white238**
I compared these changes with develop and personally saw basically no change in time on toss4 (135 seconds vs 132 seconds). This may be to how LC is setup, possibly better defaults.

@tepperly, can you see if this fixes the problem you were running into?

### CODE REVIEW

**Comment by tepperly** [comment]
LiDO doesn't generally run Serac's tests during our build/CI. My problem only happens when I run using the generic Fedora MPI on a Fedora 39 box, and it happens with LiDO executables.

File: cmake/SeracMacros.cmake [hunk 1]
```diff
@@ -150,9 +155,10 @@ macro(serac_add_tests)
                            DEPENDS_ON  ${arg_DEPENDS_ON}
                            FOLDER      serac/tests )
 
-        blt_add_test(NAME          ${test_name}
-                     COMMAND       ${test_name}
-                     NUM_MPI_TASKS ${arg_NUM_MPI_TASKS} )
+        blt_add_test(NAME            ${test_name}
+                     COMMAND         ${test_name}
+                     NUM_MPI_TASKS   ${arg_NUM_MPI_TASKS}
+                     NUM_OMP_THREADS ${arg_NUM_OMP_THREADS} )
```
Comment (tepperly) [inline cmake/SeracMacros.cmake:161]: This looks useful!

**Comment by white238** [comment]
> LiDO doesn't generally run Serac's tests during our build/CI. My problem only happens when I run using the generic Fedora MPI on a Fedora 39 box, and it happens with LiDO executables.

Ah, the `blt_add_test` macro does have the ability to set the environment OpenMP variable, it just doesnt by default.  You could utilize what I did here to set LiDO's as well. Though you may not have a wrapper macro like we do.

**Review by jamiebramwell — APPROVED** [review]

**Comment by samuelpmishLLNL** [comment]
on develop:

```
sam@provolone:~/code/serac/build-GCC-Debug$ ctest -R "qoi"
Test project /home/sam/code/serac/build-GCC-Debug
    Start 20: functional_qoi
1/1 Test #20: functional_qoi ...................   Passed   10.28 sec
```

vs this branch

```
sam@provolone:~/code/serac/build-GCC-Debug$ ctest -R "qoi"
Test project /home/sam/code/serac/build-GCC-Debug
    Start 20: functional_qoi
1/1 Test #20: functional_qoi ...................   Passed    0.29 sec

100% tests passed, 0 tests failed out of 1
```

**Review by samuelpmishLLNL — APPROVED** [review]

**Comment by samuelpmishLLNL** [comment]
it's also possible that the performance regression in openblas when running with too many threads is a bug that is fixed in more recent releases

**Pull request merged**